beacon-chain/state/stategen/getter.go

```diff
@@ -219,12 +219,24 @@ func (s *State) loadStateBySlot(ctx context.Context, slot uint64) (*state.Beacon
 		return nil, err
 	}
 
+	// In the event where start state is greater or equal to requested slot,
+	// we should just return the start state.
+	if startState.Slot() >= slot {
+		return startState, nil
+	}
+
 	// Gather the last saved block root and the slot number.
 	lastValidRoot, lastValidSlot, err := s.lastSavedBlock(ctx, slot)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get last valid block for hot state using slot")
 	}
 
+	// In the event where current state slot is greater or equal to last valid block slot,
+	// we should just process state up to input slot.
+	if startState.Slot() >= lastValidSlot {
```

Comment:
This also sounds like an impossible condition, no ? How would the state.Slot be larger than the block's slot. If there is
no block with that particular slot, its not possible to generate the state with that particular slot.

Reply:
agreed, but it's still better to be defensive here due to bugs from downstream functions. I added a comment to mention this is an impossible condition

Reply:
In that case shouldn't we fail over here and return an error ? Having something like this might mask a bigger issue.

Reply:
not necessarily, it just means the last available block slot is less than the start state slot. This serves as a short cut and also makes a lot of tests happy. 

Example:
request state with slot 200
last valid state in DB given slot index is at slot 100
last valid block in DB given slot index is <= slot 100
in this case, we can short cut and advance state from slot 100 to 200 without running rest of the logic